### PR TITLE
feat(api): Application API auth / authorization

### PR DESCRIPTION
**Pull request by jonbjarnio**
- Updated to the comment api. Simplified the database structure and updated the api layer accordingly.
- Added new auth / authorization gaurds for the application api.

### CODE REVIEW

File: apps/official-journal-application-api/src/app/app.module.ts [hunk 1]
```diff
@@ -32,13 +34,13 @@ export class AppModule implements NestModule {
...
-      })
-      .forRoutes({ path: '*', method: RequestMethod.ALL })
+      .forRoutes({ path: '/**', method: RequestMethod.ALL })
+    // .apply(WithAuthMiddleware)
+    // .exclude({
+    //   method: RequestMethod.ALL,
+    //   path: '/health',
+    //   version: '1',
+    // })
+    // .forRoutes({ path: '/**', method: RequestMethod.ALL })
```
Comment (osk) [inline apps/official-journal-application-api/src/app/app.module.ts:44]: Delete? Comment about the difference between `*` and `/**`?

File: libs/shared/middleware/src/lib/auth.middleware.ts [hunk 2]
```diff
@@ -17,6 +18,8 @@ export class WithAuthMiddleware implements NestMiddleware {
 
     const token = headers.authorization
 
+    logger.info(`Auth middleware`, { token })
```
Comment (osk) [inline libs/shared/middleware/src/lib/auth.middleware.ts:21]: Should we log the token here?

File: libs/shared/modules/src/comment/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# Comments API
```
Comment (osk) [inline libs/shared/modules/src/comment/README.md:1]: Should this have a copy of the model? I think it'll get outdated easily. Rather have some documentation about how it works together and referencing the relevant models in code?

File: libs/shared/modules/src/guards/application-auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (!nationalId) {
+      return false
+    }
+
+    request.nationalId = nationalId
+
+    // lookup user and find out if they are allowed to access the resource
+    try {
+      const userLookup = await this.applicationUserService.getUser(nationalId)
```
Comment (osk) [inline libs/shared/modules/src/guards/application-auth.guard.ts:57]: really long try catch that uses internal throws for control, refactor into other internal class methods?

**Review by osk — APPROVED** [review]

File: libs/shared/modules/src/comment/README.md (hunk 3 above)
Comment (jonbjarnio) [reply libs/shared/modules/src/comment/README.md:1]: Ah yes, I used this for myself when refactoring the api.

**Pull request merged**